src/main/scala/edu/berkeley/ce/rockslicing/InputProcessor.scala

```diff
@@ -1,50 +1,96 @@
 package edu.berkeley.ce.rockslicing
 
+import breeze.linalg.DenseVector
+
 import scala.io.Source
+import scala.util.Try
 
 object InputProcessor {
   // Processes input file: Add rock volume faces and joints to respective input list
-  def readInput(inputSource: Source): (Seq[Face], Seq[Joint]) = {
-    val lines = inputSource.getLines().toVector
-    val transitionIndex = lines.indexWhere(_ == "%")
-    assert(transitionIndex > -1)
-
-    val rockVolumeData = lines.slice(0, transitionIndex)
-    val jointData = lines.slice(transitionIndex + 1, lines.length)
-
-    val rockVolume = rockVolumeData.map { line =>
-      val tokens = line.split(" ") map(_.toDouble)
-      assert(tokens.length == 6)
-
-      val normalVec = (tokens(0), tokens(1), tokens(2))
-      val d = tokens(3)
-      val phi = tokens(4)
-      val cohesion = tokens(5)
-      Face(normalVec, d, phi, cohesion)
+  def readInput(inputSource: Source): Option[(Seq[Face], Seq[Joint])] = {
+    val lines = inputSource.getLines().zipWithIndex.toVector
+    val transitionIndex = lines.indexWhere(_._1 == "%")
+    if (transitionIndex == -1) {
+      println("Error: Input file must contain \"%\" to mark transition from rock volume to joints")
+      return None
     }
 
-    val joints = jointData.map { line =>
-      val tokens = line.split(" ") map(_.toDouble)
-      assert(tokens.length >= 11)
-      val mandatoryTokens = tokens.take(11)
-      val optionalTokens = tokens.slice(11, tokens.length)
-      assert(optionalTokens.length % 4 == 0)
+    val rockVolumeData = lines.take(transitionIndex)
+    val jointData = lines.drop(transitionIndex + 1)
+
+    val rockVolume = rockVolumeData.map { case (line, index) =>
+      // Attempt to convert each token to a Double, ignoring those that fail
+      val tokens = line.split(" ") map { x => Try(x.toDouble) } filter(_.isSuccess)
```

Comment:
Yes, that was the idea.

So if you have 11 valid doubles but there's an invalid token on the same line, it will just be ignored. This seemed sensible to me, but it may or may not be the behavior we want. Do you want me to print an error message if we find any invalid token instead of ignoring them as long as we have enough doubles?